### PR TITLE
DX-731 feat: add slack notifications config into application init flow

### PR DESCRIPTION
**Pull request by Fadhil**
## Summary

With the addition of [this](https://github.com/mindvalley/wukong-api-proxy/pull/177) feature, we can now let our devs configure slack notifications for their repo. 

In this PR, we introduce an additional prompt to enter the Slack Channel to which notifications of deployments should be sent in the `application init` flow. 

Setting a channel-name enables Slack notification for the repo with the given channel name.

Ticket: https://mindvalley.atlassian.net/browse/PXP-720

### Added
New prompt for Slack notification config in `application init`

### Output
Screenshot 2024-06-11 at 5 19 16 PM

The corresponding .wukong.toml file for the above operations:
```
[application]
name = "my-test-application"
enable = true

[application.workflows]
provider = "github_actions"
excluded_workflows = []

[[application.namespaces]]
type = "prod"

[application.namespaces.build]
build_workflow = "prod_workflow"

[application.namespaces.delivery]
target = "prod"
base_replica = 1
rollout_strategy = "rolling_upgrade"

[application.namespaces.slack_notifications]
enable = true
channel = "wukong-test-notifications"

[[application.namespaces]]
type = "staging"

[application.namespaces.build]
build_workflow = "staging_workflow"

[application.namespaces.delivery]
target = "staging"
base_replica = 1
rollout_strategy = "rolling_upgrade"

[application.namespaces.slack_notifications]
enable = true
channel = "wukong-test-notifications"

[application.addons]

```

## Notes
It is up to the developer setting these configs to ensure that the channel name exists (or will exist after setting the config) and also to add the Wukong integration to their channel if it's a private one. Wukong will attempt, and fail to send any notifications should the channel not exist, or if it hasn't been invited to a private channel.

### CODE REVIEW

**Comment by sauron-droid** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/mindvalley/wukong-cli/pull/219#" title="Author self-approved">Fadhil</a>*


The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=mindvalley%2Fwukong-cli).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/mindvalley/wukong-cli/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by jk-gan — APPROVED** [review]
LGTM

**Comment by sauron-droid** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/mindvalley/wukong-cli/pull/219#" title="Author self-approved">Fadhil</a>*, *<a href="https://github.com/mindvalley/wukong-cli/pull/219#pullrequestreview-2114578501" title="Approved">jk-gan</a>*


The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=mindvalley%2Fwukong-cli).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/mindvalley/wukong-cli/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
